### PR TITLE
v2 lint: add end to end tests

### PR DESCRIPTION
**Pull request by wolfv**
This actually uncovered a few issues :)

### CODE REVIEW

File: conda_smithy/linter/lints.py [hunk 1]
```diff
@@ -207,6 +209,20 @@ def lint_sources_should_have_hash(sources_section, lints):
...
+                if not re.match(r"[0-9a-f]{64}", source["sha256"]):
+                    lints.append(
+                        "sha256 checksum must be 64 characters long. Templates are not allowed in the sha256 checksum."
+                    )
+            if source.get("md5"):
+                if not re.match(r"[0-9a-f]{32}", source["md5"]):
+                    lints.append(
+                        "md5 checksum must be 32 characters long. Templates are not allowed in the md5 checksum."
+                    )
+
```
Comment (beckermr) [inline conda_smithy/linter/lints.py:225]: We should discuss this in the other PR still and remove it from this one.

**Review by beckermr — CHANGES_REQUESTED** [review]
Thanks for the bug fixes. One comment on not mixing PRs.

File: conda_smithy/linter/lints.py (hunk 1 above)
Comment (wolfv) [reply conda_smithy/linter/lints.py:225]: yup

Comment (wolfv) [reply conda_smithy/linter/lints.py]: removed from this PR. But we still need a release of `rattler-build-conda-compat` for these changes to go in.

**Comment by beckermr** [comment]
ok @wolfv - have fun rebasing!

**Comment by wolfv** [comment]
On it! :)

**Comment by wolfv** [comment]
OK done! :)

File: tests/test_lint_recipe.py [hunk 2]
```diff
@@ -755,12 +781,13 @@ def test_recipe_v1_test_section_with_recipe(self):
...
             self.assertIn(expected_message, lints)
 
+            # Note: v1 recipes don't have implicit "run_test.py" support
             with open(os.path.join(recipe_dir, "run_test.py"), "w") as fh:
                 fh.write("# foo")
             lints, hints = linter.lintify_meta_yaml(
                 {}, recipe_dir, recipe_version=1
             )
-            self.assertNotIn(expected_message, lints)
+            self.assertIn(expected_message, lints)
```
Comment (beckermr) [inline tests/test_lint_recipe.py:790]: There is a comment above this one that says

```
# rattler-build supports legacy run_test.py, so we need to test it
```

I cannot tell from the test + comments if we allow implicit `run_test.py` or not.

**Review by beckermr — CHANGES_REQUESTED** [review]
One item.

File: tests/test_lint_recipe.py (hunk 2 above)
Comment (wolfv) [reply tests/test_lint_recipe.py]: Sorry, removed the comment. That statement is wrong - we don't have this kind of implicit behavior.

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
LGTM and thank you!

**Pull request merged**